zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/streamprocessor/AuthorizableCommandProcessor.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.zeebe.engine.processing.streamprocessor;
+
+import io.camunda.zeebe.engine.EngineConfiguration;
+import io.camunda.zeebe.engine.processing.identity.AuthorizationCheckBehavior;
+import io.camunda.zeebe.engine.processing.identity.AuthorizationCheckBehavior.AuthorizationRequest;
+import io.camunda.zeebe.engine.processing.streamprocessor.TypedRecordProcessor.ProcessingError;
+import io.camunda.zeebe.engine.processing.streamprocessor.writers.StateWriter;
+import io.camunda.zeebe.engine.processing.streamprocessor.writers.TypedCommandWriter;
+import io.camunda.zeebe.engine.state.immutable.ProcessingState;
+import io.camunda.zeebe.protocol.impl.record.UnifiedRecordValue;
+import io.camunda.zeebe.protocol.record.RejectionType;
+import io.camunda.zeebe.protocol.record.intent.Intent;
+import io.camunda.zeebe.stream.api.records.TypedRecord;
+import io.camunda.zeebe.util.Either;
+
+/**
+ * This processor decorates processors where authorization checks are required, taking care of the
+ * authorization checks for them. If the authorization check fails, the processor will reject the
+ * command and write a rejection response back to the client.
+ *
+ * <p>This decorator should be used for processors that do not require a resource from the state to
+ * determine which resourceIds it is permitted for.
```

Comment:
❓ Is that really true? Because the process instance create processor actually fetches it from the state.

Reply:
No it's not. Good catch, I missed that during the refactoring 👍